docs/spec/glossary.rst

```diff
@@ -12,15 +12,78 @@ This section defines a few terms that may be used elsewhere in the specification
 
+   assignable
+      If a type ``B`` is "assignable to" a type ``A``, a  type checker should
+      not error on the assignment ``x: A = b``, where ``b`` is some expression
+      whose type is ``B``. Similarly for function calls and returns: ``f(b)``
+      where ``def f(x: A): ...`` and ``return b`` inside ``def f(...) -> A:``
+      are both valid (not a type error) if and only if ``B`` is assignable to
+      ``A``. In this case ``A`` is "assignable from" ``B``. For :term:`fully
+      static types <fully static type>`, "assignable to" is equivalent to
+      ":term:`subtype` of" and "assignable from" is equivalent to
+      ":term:`supertype` of". For :term:`gradual types <gradual type>`, a type
+      ``B`` is assignable to a type ``A`` if there exist fully static
+      :term:`materializations <materialize>` ``A'`` and ``B'`` of ``A`` and
+      ``B``, respectively, such that ``B'`` is a subtype of ``A'``. See
+      :ref:`type-system-concepts`.
+
+   consistent
+      Two :term:`fully static types <fully static type>` are "consistent with"
+      each other if they are the same type. Two gradual types are "consistent
+      with" each other if they could :term:`materialize` to the same type. See
+      :ref:`type-system-concepts`. If two types are consistent, they are both
+      :term:`assignable` to and from each other.
+
+   consistent subtype
+      "Consistent subtype" is synonymous with ":term:`assignable` to" (and
+      "consistent supertype" is synonymous with ":term:`assignable` from"). See
+      :ref:`type-system-concepts`.
+
    distribution
       The packaged file which is used to publish and distribute
       a release. (:pep:`426`)
 
+   fully static type
+      A type is "fully static" if it does not contain any :term:`gradual form`.
+      Fully static types represent a set of possible runtime values. Fully
+      static types participate in the :term:`subtype` relation. See
+      :ref:`type-system-concepts`.
+
+   gradual form
+      A gradual form is a :term:`type expression` which makes the type it is
+      part of not a :term:`fully static type`, but rather a representation of a
+      set of possible static types. See :ref:`type-system-concepts`. The
+      primary gradual form is :ref:`Any`. The ellipsis (`...`) is a gradual
+      form in some, but not all, contexts. It is a gradual form when used in a
+      :ref:`Callable` type, and when used in ``tuple[Any, ...]`` (but not in
+      other :ref:`tuple <tuples>` types).
+
+   gradual type
+      Types in the Python type system are "gradual". A gradual type may be a
+      :term:`fully static type`, or it may be :ref:`Any`, or a type that
+      contains ``Any`` or another :term:`gradual form`. A gradual type does not
+      necessarily represent a single set of possible runtime values; instead it
+      can represent a set of possible static types (a set of possible sets of
+      possible runtime values!). Gradual types do not participate in the
```

Comment:
I also enjoy this, but readers struggling to learn about these concepts for the first time might not 😄

```suggestion
      possible runtime values). Gradual types do not participate in the
```

Reply:
Ok, fine! 😆